library/ssl_msg.c

```diff
@@ -2048,7 +2048,7 @@ int mbedtls_ssl_fetch_input( mbedtls_ssl_context *ssl, size_t nb_want )
             if( ret < 0 )
                 return( ret );
 
-            if ( (size_t)ret > len || ( INT_MAX > SIZE_MAX && ret > SIZE_MAX ) )
+            if ( (size_t)ret > len || ( INT_MAX > SIZE_MAX && (size_t)ret > SIZE_MAX ) )
```

Comment:
Please update this one as well. It's ok to amend the last commit.

Reply:
Whoa. Did it, just absolutely forgot it was in 2 places, not 1. Amended the corresponding commit, and done the same in 2.16 branch (somehow in 2.7 I've fixed both places).